cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx

```diff
@@ -0,0 +1,372 @@
+        fetchAnnotations(): void {
+            dispatch(fetchAnnotationsAsync());
+        },
+    };
+}
+
+function convertShapesForInteractor(shapes: InteractionResult[]): number[][] {
+    const reducer = (acc: number[][], _: number, index: number, array: number[]): number[][] => {
+        if (!(index % 2)) { // 0, 2, 4
+            acc.push([
+                array[index],
+                array[index + 1],
+            ]);
+        }
+        return acc;
+    };
+
+    return shapes.filter((shape: InteractionResult): boolean => shape.shapeType === 'points')
+        .map((shape: InteractionResult): number[] => shape.points)
+        .flat().reduce(reducer, []);
+}
+
+type Props = StateToProps & DispatchToProps;
+interface State {
+    activeInteractor: Model | null;
+    activeLabelID: number;
+    interactiveStateID: number | null;
+    fetching: boolean;
+}
+
+class ToolsControlComponent extends React.PureComponent<Props, State> {
+    public constructor(props: Props) {
+        super(props);
+        this.state = {
+            activeInteractor: props.interactors.length ? props.interactors[0] : null,
+            activeLabelID: props.labels[0].id,
+            interactiveStateID: null,
+            fetching: false,
+        };
+    }
+
+    public componentDidMount(): void {
+        const { canvasInstance } = this.props;
+        canvasInstance.html().addEventListener('canvas.interacted', this.interactionListener);
+    }
+
+    public componentDidUpdate(prevProps: Props): void {
+        const { isInteraction, jobInstance } = this.props;
+        const { interactiveStateID } = this.state;
+        if (prevProps.isInteraction && !isInteraction) {
+            if (interactiveStateID !== null) {
+                jobInstance.actions.freeze(false);
+                this.setState({ interactiveStateID: null });
+            }
+        }
+    }
+
+    public componentWillUnmount(): void {
+        const { canvasInstance } = this.props;
+        canvasInstance.html().removeEventListener('canvas.interacted', this.interactionListener);
+    }
+
+    private interactionListener = async (e: Event): Promise<void> => {
+        const {
+            frame,
+            states,
+            labels,
+            jobInstance,
+            isInteraction,
+            activeLabelID,
+            fetchAnnotations,
+            updateAnnotations,
+        } = this.props;
+        const { activeInteractor, interactiveStateID } = this.state;
+
+        try {
+            this.setState({ fetching: true });
+
+            if (!isInteraction) {
+                throw Error('Canvas raises "canvas.interacted" when interaction is off');
+            }
+
+            const interactor = activeInteractor as Model;
+            const result = await core.lambda.call(jobInstance.task, interactor, {
+                task: jobInstance.task,
+                frame,
+                points: convertShapesForInteractor((e as CustomEvent).detail.shapes),
+            });
```

Comment:
As i understand, negative points in this patch works like positive, isn't it? Maybe it's better add a filter for its or something else, what do you think?

Reply:
Right now, yes.
Need to update server API for interactors (an interactor on the server should itself decide what to do with negative points)
We can filter these points for DEXTR only (till the next PR), but what problem we are trying to solve?

Reply:
It's a little bit confusing that negative points works like positive. Maybe it's better to temporary disable ability to create negative points in UI, since we have no interactors to work with and in the next PR with extended interactors functionality enable it.

Reply:
And in the next PR when we enable them, they will be confusing again.. Or maybe user will input some negative points in the future and they are ignored, what also can be confusing for him, or  a user inputs four points (two are accidentaly negative), they are filtered and dextr gets two points instead of 4. Here are a lot of open questions. If it makes you worried, we can discuss it tomorrow on the meeting